lib/ansible/modules/cloud/azure/azure_rm_rediscachefirewallrule.py

```diff
@@ -0,0 +1,317 @@
+            - The start IP address of the Azure Cache for Redis firewall rule. Must be IPv4 format.
+    end_ip_address:
+        description:
+            - The end IP address of the Azure Cache for Redis firewall rule. Must be IPv4 format.
+    state:
+      description:
+        - Assert the state of the Firewall rule of Azure Cache for Redis.
+        - Use C(present) to create or update Firewall rule of Azure Cache for Redis and C(absent) to delete it.
+      default: present
+      choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yunge Zhu(@yungezz)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create a Firewall rule for Azure Cache for Redis
+    azure_rm_rediscachefirewallrule:
+        resource_group: myResourceGroup
+        cache_name: myRedisCache
+        name: myRule
+        start_ip_address: 192.168.1.1
+        end_ip_address: 192.168.1.4
+
+  - name: Update a Firewall rule for Azure Cache for Redis
+    azure_rm_rediscachefirewallrule:
+        resource_group: myResourceGroup
+        cache_name: myRedisCache
+        name: myRule
+        end_ip_address: 192.168.1.5
+'''
+
+RETURN = '''
+id:
+    description: Id of the Azure Cache for Redis.
+    returned: always
+    type: str
+    sample:
+        "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Cache/redis/myRedis/firewallRules/myRule"
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.redis import RedisManagementClient
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+def firewall_rule_to_dict(rule):
+    return dict(
+        id=rule.id,
+        name=rule.name,
+        start_ip_address=rule.start_ip,
+        end_ip_address=rule.end_ip,
+        type=rule.type
+    )
+
+
+class Actions:
+    NoAction, CreateUpdate, Delete = range(3)
+
+
+class AzureRMRedisCacheFirewallRule(AzureRMModuleBase):
+    """Configuration class for an Azure RM Cache for Redis Firewall Rule resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            cache_name=dict(
+                type='str',
+                required=True
+            ),
+            start_ip_address=dict(
+                type='str'
+            ),
+            end_ip_address=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self._client = None
+
```

Comment:
I just wonder, maybe we should start adding "required_if"?
I think start_ip_address and end_ip_address are required when state is present?

Reply:
only one is required, the other one is not. how to express it?

Reply:
added description in doc to mention it's required when creating